hermes/bin/src/runtime_extensions/hermes/cardano/chain_follower_task.rs

```diff
@@ -0,0 +1,295 @@
+    /// Instructs the chain follower to resume generating events.
+    Continue(tokio::sync::oneshot::Sender<()>),
+}
+
+/// Chain follower handle command channel sender.
+type CommandSender = tokio::sync::mpsc::Sender<Command>;
+/// Chain follower handle command channel receiver.
+type CommandReceiver = tokio::sync::mpsc::Receiver<Command>;
+
+/// Handle used to communicate with a chain follower executor task.
+pub struct Handle {
+    /// Commands channel sender.
+    cmd_tx: CommandSender,
+}
+
+impl Handle {
+    /// Sends a command to the chain follower executor task to set its
+    /// read pointer to the given point.
+    pub fn set_read_pointer_sync(
+        &self, at: cardano_chain_follower::PointOrTip,
+    ) -> Result<Option<cardano_chain_follower::Point>> {
+        let (res_tx, res_rx) = tokio::sync::oneshot::channel();
+
+        self.cmd_tx
+            .blocking_send(Command::SetReadPointer(at, res_tx))
+            .map_err(|_| Error::InternalError)?;
+
+        res_rx
+            .blocking_recv()
+            .map_err(|_| Error::InternalError)?
+            .map_err(|_| Error::InternalError)
+    }
+
+    /// Sends a command to the chain follower executor task to stop following.
+    /// The follower continues active and following can be resumed by calling
+    /// [`Self::resume`].
+    pub fn stop(&self) -> Result<()> {
+        let (res_tx, res_rx) = tokio::sync::oneshot::channel();
+
+        self.cmd_tx
+            .blocking_send(Command::Stop(res_tx))
+            .map_err(|_| Error::InternalError)?;
+
+        drop(res_rx.blocking_recv());
+
+        Ok(())
+    }
+
+    /// Sends a command to the chain follower executor task to resume following
+    /// from the point it was previously stopped.
+    ///
+    /// This has no effect if the follower is not stopped.
+    pub fn resume(&self) -> Result<()> {
+        let (res_tx, res_rx) = tokio::sync::oneshot::channel();
+
+        self.cmd_tx
+            .blocking_send(Command::Continue(res_tx))
+            .map_err(|_| Error::InternalError)?;
+
+        drop(res_rx.blocking_recv());
+
+        Ok(())
+    }
+}
+
+/// Spawns a new Chain Follower task in the current Tokio runtime.
+pub fn spawn(
+    follower: cardano_chain_follower::Follower, app_name: HermesAppName, module_id: ModuleId,
+    chain_id: CardanoBlockchainId,
+) -> Handle {
+    let (cmd_tx, cmd_rx) = tokio::sync::mpsc::channel(1);
+
+    tokio::spawn(super::chain_follower_task::executor(
+        cmd_rx, follower, app_name, module_id, chain_id,
+    ));
+
+    Handle { cmd_tx }
+}
+
+/// Runs a Cardano chain follower that generates events for the given application module
+/// and is connected to the given chain.
+#[instrument(skip(cmd_rx, follower), fields(app_name = %app_name, module_id = %module_id))]
+async fn executor(
+    mut cmd_rx: CommandReceiver, mut follower: cardano_chain_follower::Follower,
+    app_name: HermesAppName, module_id: ModuleId, chain_id: CardanoBlockchainId,
+) {
```

Comment:
Can we consider replacing ` 'exec_loop: loop` with something like `while !stopped`(_see the draft below_)?

* **Readability and Intent** 
Using `while !stopped` immediately conveys the loop's continuation condition directly in the loop's declaration, making it clear to readers that the loop will continue executing as long as `stopped` is `false`. This directly ties the loop's execution to the state of a specific variable (stopped), making the code easier to understand.

* **Maintains Focus on Business Logic**
This can simplify the loop control structure and allow future maintainers or readers to focus more on the business
 logic within the loop rather than the mechanics of loop control. 
 
*  **Improves Code Clarity**
The `while !stopped` loop makes the control logic dependent on the state of the stopped variable very explicit. It eliminates the need for readers to identify where and how the loop might be terminated. This direct association between the loop continuation and the stopped variable enhances clarity by immediately making the loop's termination condition visible.

```rust
#[instrument(skip(cmd_rx, follower), fields(app_name = %app_name, module_id = %module_id))]
async fn executor(
    mut cmd_rx: CommandReceiver, mut follower: cardano_chain_follower::Follower,
    app_name: HermesAppName, module_id: ModuleId, chain_id: CardanoBlockchainId,
) {
    let network = chain_id.into();
    let module_state_key = (app_name, module_id, network);

    let mut stopped = false;

    while !stopped {
        tokio::select! {
            res = cmd_rx.recv() => {
                // Improved pattern matching for clarity.
                match res {
                    Some(cmd) => {
                        match cmd {
                            Command::SetReadPointer(follow_from, res_tx) => {
                                // Handling the potential error from set_read_pointer asynchronously.
                                if let Err(e) = set_read_pointer(&mut follower, follow_from).await {
                                    eprintln!("Error setting read pointer: {:?}", e);
                                    // Consider if stopping on error is desired.
                                }
                                // Now explicitly ignoring the send result but consider logging the error.
                                let _ = res_tx.send(stopped);
                            }
                            Command::Stop(res_tx) => {
                                stopped = true;
                                let _ = res_tx.send(());
                            }
                            Command::Continue(res_tx) => {
                                stopped = false;
                                let _ = res_tx.send(());
                            }
                        }
                    },
                    None => break, // Cleaner break from the loop if channel is closed.
                }
            }

            result = follower.next(), if !stopped => {
                // Error handling for process_chain_update is similar to set_read_pointer.
                if let Err(e) = process_chain_update(&module_state_key, chain_id, result).await {
                    eprintln!("Error processing chain update: {:?}", e);
                    // Consider if stopping on error is desired.
                }
            }
        }
    }

    // Improved resource cleanup with error handling.
    if let Err(e) = follower.close().await {
        eprintln!("Error closing the follower: {:?}", e);
    }
}

```

Reply:
I this case we don't want to stop the loop if `stopped = false`. 

`stopped` will be `false` when the app module that is using this follower wants to stop the Cardano Runtime Extension event generation. In that case, we still want to wait for commands without progressing the following process.

The loop will only be broken if there are no command `Sender`s or the module's subscription state is removed from the extension's state.

I used a label to name the loop and make it clear when using `break`.

Reply:
Thank you, @FelipeRosa, for the clarifications.